docs/core/deploying/trimming-options.md

```diff
@@ -16,10 +16,14 @@ Trimming with `PublishTrimmed` was introduced in .NET Core 3.0. The other option
 
 - `<PublishTrimmed>true</PublishTrimmed>`
 
-   Enable trimming during publish, with the default settings defined by the SDK.
+   Enable trimming during publish. This also turns off trim-incompatible features and enables [trim analysis](#roslyn-analyzer) during build.
+
+Place this in the project file to ensure that the setting applies during `dotnet build`, not just `dotnet publish`.
 
 This will trim any assemblies which have been configured for trimming. With `Microsoft.NET.Sdk` in .NET 6, this includes the any assemblies with `[AssemblyMetadata("IsTrimmable", "True")]`, which is the case for framework assemblies. In .NET 5, framework assemblies from the netcoreapp runtime pack are configured for trimming via `<IsTrimmable>` MSBuild metadata. Other SDKs may define different defaults.
 
+Starting in .NET 6, this setting also enables the trim compatibility [Roslyn analyzer](#roslyn-analyzer), and disables [features](#trimming-framework-library-features) which are incompatible with trimming.
```

Comment:
Should we update the features list to include the new ones like BuiltInComInteropSupport, EnableCppCLIHostActivation etc.?

Reply:
I don't think we should document the ones that are already disabled by default when linking - we don't want to encourage people to enable the unsupported ones.

But it looks like `UseNativeHttpHandler` is missing from the list. @eerhardt do you think we should document that one? It looks to me like `UseNativeHttpHandler=true` will prevent some (but not all) references to `HttpClientHandler` - I'm not sure whether this feature was designed for trimming in particular.

Reply:
I do find it a bit confusing that we are explicitly saying `disables features [link] which are incompatible with trimming`, but then the link doesn't list any features which are incompatible with trimming.

Maybe we should have 2 sections? Features that can be toggled when trimming, and features that are not compatible with trimming?

> But it looks like UseNativeHttpHandler is missing from the list.

@EgorBo @akoeplinger @steveisok @marek-safar - should we be documenting `UseNativeHttpHandler` here? I believe it is only valid for Android and iOS apps.

Reply:
I added  a section for the trim-incompatible features, and also added a few more switches to the optional features section, including UseNativeHttpHandler.